eng/tools/RepoTasks/RepoTasks.csproj

```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net7.0</TargetFrameworks>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework)</TargetFrameworks>
```

Comment:
Changes here cannot work until we have an 8.0.0 SDK containing `net8.0` bits. The basic idea is `RepoTasks` and `GenerateFiles` aren't part of the product. They're instead used in `msbuild` (or `dotnet msbuild`) and to generate the extra Directory.Build.props, Directory.Build.targets, and dotnet-tools.json files used in almost every other project in the repo.

Why did this file and RepoTasks.tasks need to change in the first place❔ Note `GenerateFiles` still targets `net5.0` and (other than the deprecated TFM warning we're suppressing somewhere) that's fine.

See for example
``` text
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

Reply:
Yep, that's why this is blocked on https://github.com/dotnet/installer/pull/15151. It needs to change because SourceBuild breaks if it doesn't target the latest TFM (though I don't know why the same isn't true for `GenerateFiles`) - https://github.com/dotnet/installer/pull/15151#issuecomment-1359316105

Reply:
> though I don't know why the same isn't true for GenerateFiles

@dougbu @wtgodbe, GenerateFiles.csproj doesn't need to use the latest TFM because it doesn't build during source-build: https://github.com/dotnet/aspnetcore/blob/cefc6ccf4b23ac9f07418cd10e374cf533276eee/eng/tools/GenerateFiles/GenerateFiles.csproj#L5